### ISSUE TITLE
[FEAT]: Improve UX to let user know which LLM is actually going to be used

### ISSUE THREAD

**Issue by kcalliauw** [Mintplex-Labs/anything-llm#2642]
### What would you like to see?

As a user I tend to like to switch between LLMs often (this includes switching between Ollama models and OpenAI + Anthropic external models), and I currently find it very confusing / impossible to predict which of my LLMs are going to be chosen for a thread.

- LLM can be set in the Instance settings 
- LLM can be set in the workspace settings
- LLM can be set in the workspace settings (but specific for agents)

As far as I can tell, if I change the chat agent in my workspace, the existing threads remain unaffected, which "feels" like a bug to me. I would expect (where possible) context to be carried over to the newly chosen LLM.

Ideally, like in various apps: ChatGPT or TypingMind, the LLM can be chosen inside the chat context/thread. For example, here are a few UX experiences I find to be very beneficial:

![CleanShot 2024-11-18 at 09 35 41](https://github.com/user-attachments/assets/8b6b60b9-7256-4fe5-9a22-b194b523f3ba)

![CleanShot 2024-11-18 at 09 36 44](https://github.com/user-attachments/assets/ec51fabb-f22c-4d62-8811-bc16bc62a42f)

Just my 2 cents. Keep up the great work!

**Comment by xiajun2001** [comment]
waiting for it,too~

**Comment by timothycarambat** [comment]
> if I change the chat agent in my workspace, the existing threads remain unaffected

This is not true (or certainly should not be). How are you verifying this is the case? 
_please don't say you are asking the LLM what model it is 🤞_

**Comment by timothycarambat** [comment]
You can now do this in the prompt window by click on the "brain icon" to swap to any model/provider combo
